### PR TITLE
[refactor] make AsyncPipe its own class

### PR DESCRIPTION
**Pull request by msbaines**
Will separate out the implementations in a follow-up PR.

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: fairscale/nn/pipe/async_pipe.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+from .multiprocess_pipe import MultiProcessPipe
+from .types import PipelineStyle
+
+
+class AsyncPipe(MultiProcessPipe):
```
Comment (sidgoyal78) [inline fairscale/nn/pipe/async_pipe.py:10]: Are you planning to separate out Async related stuff from pipeline.py as well in this PR?

**Review by sidgoyal78 — COMMENTED** [review]
Nice refactor, thanks for the PR.

File: fairscale/nn/pipe/async_pipe.py (hunk 1 above)
Comment (min-xu-ai) [inline fairscale/nn/pipe/async_pipe.py:10]: add a class docstring?

Also, curious, why type: ignore is needed for its init?

**Review by min-xu-ai — APPROVED** [review]
looks really nice for this separation and simplification. big fan of this!

File: fairscale/nn/pipe/async_pipe.py (hunk 1 above)
Comment (msbaines) [reply fairscale/nn/pipe/async_pipe.py:10]: Planning to do that in a followup. Too much to take on in one PR.

Comment (msbaines) [reply fairscale/nn/pipe/async_pipe.py:10]: I think because I'm not directly handling the args.

**Pull request merged**